### PR TITLE
feat(bort): Add new Metrics Overview page, improve metrics wording

### PR DESCRIPTION
**Pull request by Blisse**
deploy-preview-693--memfault-docs netlify app_docs_android_android-metrics

Screenshot 2023-11-30 at 00 31 56

### CODE REVIEW

File: docs/android/android-custom-metrics.mdx [hunk 1]
```diff
@@ -42,13 +36,11 @@ Fleet Sampling resolution.
...
 
-Bort aggregates each metric at the end of the reporting period (usually 1 hour).
-These aggregates are:
-
-- Shown on the device timeline.
-- Available for use as [Timeseries](/docs/platform/metrics#timeseries) or
-  [Attributes](/docs/platform/metrics#attributes) metrics, aggregated across the
-  entire fleet.
+The Reporting library also supports aggregating collected values to produce a
+single metric in the heartbeat report at the each of the heartbeat interval
```
Comment (ejohnso49) [inline docs/android/android-custom-metrics.mdx:40]:
```suggestion
single metric in the heartbeat report at the end each of the heartbeat interval
```

File: docs/android/android-custom-metrics.mdx [hunk 2]
```diff
@@ -57,6 +49,17 @@ Aggregate metrics are supported from Bort 4.0.0 onwards. This is part of the
 
 :::
 
+For example, if you record 60 temperature values, one for every minute in an
+hour, using the Reporting library as specified below, Memfault will show all 60
+values on the Device Timeline in HRT, and produce 2 heartbeat metrics
+(`device_temp.min`, `device_temp.max`), as a result of the MIN and MAX
+aggregations, which represent the min and max values recorded over the hour.
```
Comment (ejohnso49) [inline docs/android/android-custom-metrics.mdx:56]: I would split this up into multiple sentences

File: docs/android/android-custom-metrics.mdx [hunk 3]
```diff
@@ -17,21 +17,29 @@ Memfault will collect various AOSP metrics automatically once integrated, like
...
+`MAX` aggregations.
 
-Individual metric events are uploaded and displayed in high resolution on the
-device timeline. This will appear similar to the existing batterystats metrics.
+### High Resolution Telemetry (HRT) {#high-resolution-telemetry-hrt}
 
-All metrics are uploaded as part of HRT, even if there are no aggregations
-requested.
+The Reporting library will upload and display all collected values in high
+resolution on the device timeline in every heartbeat report. These HRT metrics
```
Comment (sjp4) [inline docs/android/android-custom-metrics.mdx:41]: HRT is not technically part of the heartbeat report (although it it collected at the same time). Not sure if that matters... but I'm hesitant to link them too closely.

File: docs/android/android-custom-metrics.mdx [hunk 4]
```diff
@@ -17,21 +17,29 @@ Memfault will collect various AOSP metrics automatically once integrated, like
...
+are not available for use as Timeseries or Attributes metrics.
```
Comment (sjp4) [inline docs/android/android-custom-metrics.mdx:42]: The wording here could be confusing - maybe we shouldn't say `HRT metrics` (less overloading of the word `metric`)? Maybe something like `All metrics are shown in high resolution on the timeline, and any aggregations are available for use as Timeseries metrics or Attributes`? (`Attributes metrics` doesn't make sense)

File: docs/android/android-custom-metrics.mdx [hunk 5]
```diff
@@ -40,15 +48,14 @@ Fleet Sampling resolution.
...
-These aggregates are:
+### Metric Aggregrations {#aggregate-metrics}
 
-- Shown on the device timeline.
-- Available for use as [Timeseries](/docs/platform/metrics#timeseries) or
-  [Attributes](/docs/platform/metrics#attributes) metrics, aggregated across the
-  entire fleet.
+The Reporting library supports performing pre-set aggregations on reported
+values, to produce a single aggregate metric in the heartbeat report, at the end
+of the heartbeat interval (usually 1 hour). This single metric is also shown on
```
Comment (sjp4) [inline docs/android/android-custom-metrics.mdx:55]: It's not shown on the timeline when HRT is available.

File: docs/android/android-metrics-overview.mdx [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+referred to as a "heartbeat" metric, derived during the "heartbeat" interval,
+and uploaded in the "heartbeat" report.
+
+<ImageFigure src="/img/docs/android/metrics-heartbeat.png" alt="HRT example">
+  Only 1 value is recorded every hour.
+</ImageFigure>
+
+### Attributes {#attributes}
+
+Marking a heartbeat metric as an **Attribute** implies that we primarily care
```
Comment (sjp4) [inline docs/android/android-metrics-overview.mdx:24]: Show a screenshot of the attributes tab here

File: docs/android/android-metrics-overview.mdx [hunk 7]
```diff
@@ -0,0 +1,82 @@
+---
+id: android-metrics-overview
+title: Android Metrics Overview
+sidebar_label: Metrics Overview
+---
+
+Memfault's Bort SDK makes it easy to collect all kinds of metrics to diagnose
+issues on a per-device basis, and to monitor trends on a fleet-wide basis.
```
Comment (sjp4) [inline docs/android/android-metrics-overview.mdx:8]: Before diving into the different metric types: maybe link to the generic metrics documentation which describes how to configure each individual metric as timeseries/attribute/etc (and describes the quotas?)?

File: docs/android/android-metrics-overview.mdx [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+Marking a heartbeat metric as an **Attribute** implies that we primarily care
+about the latest value of the metric, for that device. This allows for searching
+for all current devices matching that Attribute. We can also create a Device Set
+for that metric, which gives a coarse-grained daily count of devices matching
+the attribute condition.
+
+### Timeseries {#timeseries}
+
+On the other hand, marking a heartbeat metric as **Timeseries** implies that we
```
Comment (sjp4) [inline docs/android/android-metrics-overview.mdx:32]: Screenshot of a fleet-side chart?

File: docs/android/android-metrics-overview.mdx [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+visualized in more detail in the Device Timeline view. The HRT metrics are
+uploaded alongside the heartbeat report.
+
+<ImageFigure src="/img/docs/android/metrics-hrt.png" alt="HRT example">
+  Multiple values per hour can be recorded and visualized using HRT.
+</ImageFigure>
+
+Note that these HRT metrics are **not** available for querying as a metric, and
+not available for use as Timeseries or Attribute metrics. They are used
+primarily for device-level diagnostics.
```
Comment (sjp4) [inline docs/android/android-metrics-overview.mdx:55]: s/primarily/exclusively

`diagnostics & debugging`?

File: docs/android/android-metrics-overview.mdx [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+- device temperature
+- network connectivity
+- network usage
+
+### Custom metrics {#custom-metrics}
+
+_Main article: [Custom Metrics](/docs/android/android-custom-metrics)_
+
+The Bort SDK also allows for collecting your own domain-specific metrics in
+addition to the built-in metrics, via the use of the Reporting library.
```
Comment (sjp4) [inline docs/android/android-metrics-overview.mdx:76]: `via the use of the Reporting library from any kotlin/java/native code on the device`?

File: docs/android/android-custom-metrics.mdx (hunk 4 above)
Comment (Blisse) [reply docs/android/android-custom-metrics.mdx:42]: Yeah I want to rip apart this part of our docs. The problem is that we have an Aggregate Metrics section right after, so I'm trying to differentiate between the 2 concepts, but in reality they're like intrinsically linked.

File: docs/android/android-custom-metrics.mdx (hunk 5 above)
Comment (Blisse) [reply docs/android/android-custom-metrics.mdx:55]: Ahh the metrics I was looking at weren't collected by Reporting lib

File: docs/android/android-metrics-overview.mdx (hunk 7 above)
Comment (Blisse) [reply docs/android/android-metrics-overview.mdx]: I ended up linking to the general Metrics page. I want to think through some organizational changes to that Metrics docs (the Timeseries section has like 1 short blurb and no examples).

File: docs/android/android-custom-metrics.mdx (hunk 4 above)
Comment (Blisse) [reply docs/android/android-custom-metrics.mdx]: I ended up just with 

```
Every value recorded by the Reporting library is uploaded and displayed in high
resolution on the device timeline.

These values are not available for use as Timeseries or Attribute metrics.
```

The next section and the previous section already covers aggregations so I didn't go that direction. 

Maybe we could merge the Debugging Resolution part and explain they won't be shown in HRT if the resolution is off?

**Review by sjp4 — APPROVED** [review]

**Pull request merged**